### ISSUE TITLE
Coding Question about K2_min_quantize_loss_layer.cpp

### ISSUE THREAD

**Issue by shuokay** [kevinlin311tw/cvpr16-deepbit#1]
I find [this line](https://github.com/kevinlin311tw/cvpr16-deepbit/blob/master/src/caffe/layers/K2_min_quantize_loss_layer.cpp#L42) is different to your paper.
According to your paper, I think it should be 
```cpp
loss = loss + (binary[i]-0.5-bottom_data[i])*(binary[i]-0.5-bottom_data[i]);
```
Can you tell me where is I am wrong?

**Comment by kevinlin311tw** [comment]
Note that we implement the sign function as `sgn(k) = 1 if k > 0, and 0 otherwise`, which is different from the definition `sgn(k) = 1 if k > 0, and -1 otherwise` in the paper. Thus, we can rewrite equation(1) as `b = sgn(F(x; W))`

**Comment by icodingc** [comment]
[EvenDistLoss](https://github.com/kevinlin311tw/cvpr16-deepbit/blob/master/src/caffe/layers/K1_even_dist_loss_layer.cpp#L15)
I want to know why there are two bottoms.

**Comment by kevinlin311tw** [comment]
I forgot to remove bottom[1]. Because we didn't use bottom[1] to compute loss, you can ignore it.

**Comment by icodingc** [comment]
ok. thank you.

**Comment by shuokay** [comment]
Thanks @kevinlin311tw
